src/components/Fields/TextField.scss

```diff
@@ -128,44 +128,37 @@ $textFieldLabelColor: $charcoal !default;
         margin-left: 5px;
     }
 
-    .uir-text-field-left-icon {
+    .uir-text-field-left-icon,
+    .uir-text-field-right-icon {
         position: absolute;
-        top: 24px;
-        left: 2px;
+        top: 21px;
         line-height: 1em;
     }
 
+    .uir-text-field-left-icon {
+        left: 2px;
+    }
+
     .uir-text-field-right-icon {
-        position: absolute;
-        top: 21px;
         right: 2px;
-        line-height: 1em;
 
         .uir-tooltip-contents {
             line-height: normal;
         }
+    }
 
-        .uir-button {
-            position: relative;
-            box-sizing: border-box;
-            display: inline-block;
-            width: 16px;
-            min-width: 16px;
-            height: 16px;
-            padding: 0;
-            margin: 0;
-
-            &-content {
-                line-height: inherit;
-            }
+    .uir-button.uir-text-field-icon-wrapper {
+        min-width: 16px;
+        padding: 0;
+        margin: 0;
 
-            &:hover {
-                background-color: transparent;
-            }
+        &:focus,
+        &:hover {
+            background-color: transparent;
+        }
 
-            .uir-icon-clear:hover {
-                fill: $charcoal;
-            }
+        .uir-icon-clear:hover {
+            fill: $charcoal;
```

Comment:
It'd be nice to add `:focus` to this as well. Currently there is no indication of focus when I tab into the clear icon.